core/client/src/blockchain.rs

```diff
@@ -197,21 +197,11 @@ impl<Block: BlockT> TreeRoute<Block> {
 }
 
 /// Compute a tree-route between two blocks. See tree-route docs for more details.
-pub fn tree_route<Block: BlockT, Backend: HeaderBackend<Block>>(
-	backend: &Backend,
+pub fn tree_route<Block: BlockT, F: Fn(BlockId<Block>) -> Result<<Block as BlockT>::Header>>(
+	load_header: F,
 	from: BlockId<Block>,
 	to: BlockId<Block>,
 ) -> Result<TreeRoute<Block>> {
-	use sr_primitives::traits::Header;
-
-	let load_header = |id: BlockId<Block>| {
```

Comment:
Maybe we should provide this as utility function? To not have the same code copied over and over again?

Reply:
how  would one do that without reintroducing the binding to the `HeaderBackend` trait, that - if I recall correctly - not all callers actually implement....